Examples/Scripts/Python/MLAmbiguityResolution/ambiguity_solver_full_chain.py

```diff
@@ -0,0 +1,338 @@
+import glob
+import os
+import math
+
+import pandas as pd
+import numpy as np
+
+from sklearn.preprocessing import LabelEncoder, StandardScaler, OrdinalEncoder
+from sklearn.cluster import DBSCAN
+
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.utils
+
+import ast
+import onnxruntime as ort
+
+
+def PrepareDataSet(CKS_files: list[str]) -> pd.DataFrame:
+    """Read the dataset from the different file, remove the pure duplicate tracks and combine the datasets"""
+    """
+    @param[in] CKS_files: DataFrame contain the data from each track files (1 file per events usually)
+    @return: combined DataFrame containing all the track, ordered by events and then by truth particle ID in each event 
+    """
+    globalindex = 0
+    data = []
+    for f in CKS_files:
+        datafile = pd.read_csv(f)
+        # Remove tracks with less than 7 measurements
+        datafile = datafile[datafile["nMeasurements"] > 6]
+        datafile = datafile.sort_values("good/duplicate/fake", ascending=False)
+        # Remove pure duplicate (tracks purely identical) keep the ones good one if among them.
+        datafile = datafile.drop_duplicates(
+            subset=[
+                "particleId",
+                "Hits_ID",
+                "nOutliers",
+                "nHoles",
+                "nSharedHits",
+                "chi2",
+            ],
+            keep="first",
+        )
+        datafile = datafile.sort_values("particleId")
+        # Set truth particle ID as index
+        datafile = datafile.set_index("particleId")
+        # Transform the hit list from a string to an actual list
+        hitsIds = []
+        mergedIds = []
+        for list in datafile["Hits_ID"].values:
+            hitsIds.append(ast.literal_eval(list))
+        datafile["Hits_ID"] = hitsIds
+        # Combine dataset
+        data.append(datafile)
+    return data
+
+
+def PrepareData(data: pd.DataFrame) -> tuple[np.ndarray, np.ndarray]:
+    """Prepare the data"""
+    """
+    @param[in] data: input DataFrame to be prepared
+    @return: array of the network input and the corresponding truth  
+    """
+    target_column = "good/duplicate/fake"
+    global avg_mean
+    global avg_sdv
+    # Separate the truth from the input variables
+    y = LabelEncoder().fit(data[target_column]).transform(data[target_column])
+    # Remove truth and useless variable
+    input = data.drop(
+        columns=[
+            target_column,
+            "track_id",
+            "nMajorityHits",
+            "nSharedHits",
+            "truthMatchProbability",
+            "Hits_ID",
+            "chi2",
+            "pT",
+            "cluster",
+        ]
+    )
+    # Normalise variables
+    x_cat = OrdinalEncoder().fit_transform(input.select_dtypes("object"))
+    x = np.concatenate((x_cat, input), axis=1)
+    return x, y
+
+
+def ClusterTracks(
```

Comment:
personally I am not a fan of functions/variables starting with a capital letter. sorry had to complain at least once :smile:

Reply:
you have also the official ACTS coding standard on your side :D https://acts.readthedocs.io/en/latest/codeguide.html#n-3-functions-and-methods-use-mixedcase
But not sure how strict we are in this respective

Reply:
It hurt my brain to not start a line with capital letter but you are 100% right I will go change this

Reply:
Should be done now, I hope I haven't missed one